ui/src/app/pages/data/data-set/dataset-page.tsx

```diff
@@ -605,12 +604,37 @@ const DataSetPage = fp.flow(withUserProfile(), withCurrentWorkspace(), withUrlPa
       this.setState({selectedDomainValuePairs: valuesSelected, dataSetTouched: true});
     }
 
-    get selectAll() {
-      return fp.isEmpty(this.state.selectedDomainValuePairs);
+    // Append newValueSets (values from API) to state's valueSets and to selectedDomainValuePairs.
+    // Set valuesLoading to false once the state is updated
+    updateValueSets(newValueSets) {
+      const {selectedDomainValuePairs, valueSets} = this.state;
+
+      newValueSets.map(newValueSet => {
+        newValueSet.values.items.map(value => {
+          selectedDomainValuePairs.push({domain: newValueSet.domain, value: value.value});
+        });
+      });
+      this.setState({
+        valueSets: valueSets.concat(newValueSets),
+        selectedDomainValuePairs: selectedDomainValuePairs,
+        valuesLoading: false
+      });
+    }
+
+    // Returns true if selected values set is empty or is not equal to the total values displayed
+    get partialOrNoneValuesSelected() {
+      return fp.isEmpty(this.state.selectedDomainValuePairs) ||
+          this.state.selectedDomainValuePairs.length !== this.valuesCount;
+    }
+
+    get valuesCount() {
+      let count = 0 ;
+      this.state.valueSets.map(value => count += value.values.items.length);
+      return count;
     }
 
     selectAllValues() {
-      if (!this.selectAll) {
+      if (!this.partialOrNoneValuesSelected) {
```

Comment:
(Opt) A common piece of coding advice I've heard is that it's usually better to have if-else statements based on a positive conditional rather than a negated one. In this case, it may actually be simpler to have a method called .isAllSelected, which would let you write something like:

if (this.isAllSelected) {
  this.setState({...});
}

If that matches the intent of the code, it would be much easier to read than the current somewhat-complicated condition.